dynamorm/model.py

```diff
@@ -283,15 +278,56 @@ def scan(cls, scan_kwargs=None, **kwargs):
             Thing.scan(address__state="CA")
             Thing.scan(address__state__begins_with="C")
 
+        This returns a generator, which will continue to yield items until all matching the scan are produced,
+        abstracting away pagination. More information on scan pagination: http://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Scan.html#Scan.Pagination
+
         :param dict scan_kwargs: Extra parameters that should be passed through to the Table scan function
         :param \*\*kwargs: The key(s) and value(s) to filter based on
         """  # noqa
-        resp = cls.Table.scan(scan_kwargs=scan_kwargs, **kwargs)
-        return [
-            cls.new_from_raw(raw)
-            for raw in resp['Items']
-            if raw is not None
-        ]
+        return cls._yield_items('scan', dynamo_kwargs=scan_kwargs, **kwargs)
+
+    @classmethod
+    def _yield_items(cls, method_name, dynamo_kwargs=None, **kwargs):
+        """Private helper method to yield items from a scan or query response
+
+        :param method_name: The cls.Table.<method_name> that should be called (one of: 'scan','query'))
+        :param dict dynamo_kwargs: Extra parameters that should be passed through from query_kwargs or scan_kwargs
+        :param \*\*kwargs: The key(s) and value(s) to filter based on
+        """
+        if method_name == 'scan':
+            method = cls.Table.scan
+            dynamo_kwargs_key = 'scan_kwargs'
+        elif method_name == 'query':
+            method = cls.Table.query
+            dynamo_kwargs_key = 'query_kwargs'
+        else:
+            raise ValueError('Method should be one of: scan,query')
```

Comment:
This is an internal method, so EAFP...

```
method = getattr(cls.Table, method_name)
dynamo_kwargs_key  = '_'.join([method_name, 'kwargs'])
```

The `getattr` will blow up if you pass a method that doesn't exist.